exporter/datadogexporter/internal/hostmetadata/metadata_test.go

```diff
@@ -190,13 +190,13 @@ func TestPushMetadata(t *testing.T) {
 		assert.Equal(t, "apikey", r.Header.Get("DD-Api-Key"))
 		assert.Equal(t, "otelcontribcol/1.0", r.Header.Get("User-Agent"))
 		reader, err := gzip.NewReader(r.Body)
-		require.NoError(t, err)
+		assert.NoError(t, err)
```

Comment:
I'm a bit confused by the change. We already had a linter that was forcing the use of require when checking errors, sometimes incorrectly (eg you don't want require in deferred functions). But then in this case require was correct because we don't want to proceed checking return values if there was an error. 

So 
- is the linter telling that this line needs to be fixed?
- what are we changing about the linter, since we already had one